Add LightShow

MagoKimbra opened the pull request:
Add partition for LightShow and insert new menu LightShow for read and write file.

MagoKimbra committed:
Update index.html

Inster Lightshow

MagoKimbra committed:
Update index.html

Add LightShow

MagoKimbra committed:
Update index.html

Add Boombox

MagoKimbra committed:
Update filebrowser.html

Add Boombox

MagoKimbra committed:
Update filebrowser.css

MagoKimbra committed:
Update index.html

MagoKimbra committed:
Update configure-web.sh

MagoKimbra committed:
Update teslausb_setup_variables.conf.sample

MagoKimbra committed:
Update install.sh

MagoKimbra committed:
Update configure-web.sh

MagoKimbra committed:
Update create-backingfiles.sh

MagoKimbra committed:
Update setup-teslausb

MagoKimbra committed:
Update setup-teslausb

MagoKimbra committed:
Update filebrowser.css

MagoKimbra committed:
Update filebrowser.html

MagoKimbra committed:
Update index.html

MagoKimbra committed:
Update install.sh

MagoKimbra committed:
Update auto.www

MagoKimbra committed:
Update auto.www

MagoKimbra committed:
Update enable_gadget.sh

MagoKimbra committed:
Update disable_gadget.sh

MagoKimbra committed:
Update enable_gadget.sh

MagoKimbra committed:
Update enable_gadget.sh

MagoKimbra committed:
Update enable_gadget.sh

MagoKimbra committed:
Update index.html

MagoKimbra committed:
Update index.html

MagoKimbra committed:
Update config.sh

MagoKimbra committed:
Update index.html

MagoKimbra committed:
Update index.html

MagoKimbra committed:
Update teslausb_setup_variables.conf.sample

marcone commented on pi-gen-sources/00-teslausb-tweaks/files/teslausb_setup_variables.conf.sample:
```diff
@@ -90,10 +90,13 @@ export SHARE_PASSWORD='password'
-#export BOOMBOX_SIZE=100M
+export CAM_SIZE=80%
+export MUSIC_SIZE=15%
+export LIGHTSHOW_SIZE=2%
```
This line should be the only change in the sample config file, and it should use a fixed size instead of a percentage. The rest of the file should remain unchanged.

marcone commented on setup/generic/install.sh:
```diff
@@ -63,7 +63,7 @@ then
 		#!/bin/bash
 		{
-		  while ! curl -s https://raw.githubusercontent.com/marcone/teslausb/main-dev/setup/generic/install.sh
+		  while ! curl -s https://raw.githubusercontent.com/MagoKimbra/teslausb/main-dev/setup/generic/install.sh
```
Don't change the repo

MagoKimbra replied:
But then here when I test it searches mine or does it go to yours and take the old files?
Then it's okay that the right name remains in yours, but now while I'm testing I need you to download the files I modified.

marcone commented on setup/generic/install.sh:
```diff
@@ -95,7 +95,7 @@ then
 
     {
-      while ! curl -s https://raw.githubusercontent.com/marcone/teslausb/main-dev/tools/debian-resizefs.sh
+      while ! curl -s https://raw.githubusercontent.com/MagoKimbra/teslausb/main-dev/tools/debian-resizefs.sh
```
Don't change the repo

marcone commented on setup/generic/install.sh:
```diff
@@ -127,7 +127,7 @@ fi
 if [ ! -e /boot/teslausb_setup_variables.conf ] && [ ! -e /root/teslausb_setup_variables.conf ]
 then
-  while ! curl -o /boot/teslausb_setup_variables.conf https://raw.githubusercontent.com/marcone/teslausb/main-dev/pi-gen-sources/00-teslausb-tweaks/files/teslausb_setup_variables.conf.sample
+  while ! curl -o /boot/teslausb_setup_variables.conf https://raw.githubusercontent.com/MagoKimbra/teslausb/main-dev/pi-gen-sources/00-teslausb-tweaks/files/teslausb_setup_variables.conf.sample
```
Don't change the repo

MagoKimbra replied:
As before, while I am testing it must download the files I modified.

marcone commented on setup/generic/install.sh:
```diff
@@ -136,7 +136,7 @@ fi
 if [ ! -e /boot/wpa_supplicant.conf.sample ]
 then
-  while ! curl -o /boot/wpa_supplicant.conf.sample https://raw.githubusercontent.com/marcone/teslausb/main-dev/pi-gen-sources/00-teslausb-tweaks/files/wpa_supplicant.conf.sample
+  while ! curl -o /boot/wpa_supplicant.conf.sample https://raw.githubusercontent.com/MagoKimbra/teslausb/main-dev/pi-gen-sources/00-teslausb-tweaks/files/wpa_supplicant.conf.sample
```
Don't change the repo

marcone commented on setup/generic/install.sh:
```diff
@@ -146,7 +146,7 @@ fi
 # for setting up a Raspberry Pi with the prebuilt image
 rm -f /etc/rc.local
-while ! curl -o /etc/rc.local https://raw.githubusercontent.com/marcone/teslausb/main-dev/pi-gen-sources/00-teslausb-tweaks/files/rc.local
+while ! curl -o /etc/rc.local https://raw.githubusercontent.com/MagoKimbra/teslausb/main-dev/pi-gen-sources/00-teslausb-tweaks/files/rc.local
```
Don't change the repo

marcone commented on setup/generic/install.sh:
```diff
@@ -189,17 +189,17 @@ then
+	  echo "| /boot/teslausb_setup_variables.conf with your favorite                    |"
+	  echo "| editor, e.g. 'nano /boot/teslausb_setup_variables.conf' and fill in       |"
+	  echo "| the required variables. Instructions are in the file, and at              |"
+	  echo "| https://github.com/MagoKimbra/teslausb/blob/main-dev/doc/OneStepSetup.md  |"
```
Don't change the repo (and then you don't need to change any of the surrounding lines either)

marcone commented on setup/pi/create-backingfiles.sh:
```diff
@@ -216,6 +221,15 @@ then
 
 REMAINING_SPACE="$(available_space)"
+
+if [ "$REMAINING_SPACE" -ge 1024 ] && [ "$LIGHTSHOW_DISK_SIZE" -gt 0 ]
```
This check seems wrong. It's checking that there is more than 1MB available, but it seems likely that LIGHTSHOW_DISK_SIZE will be greater than 1MB

marcone commented on teslausb-www/html/filebrowser.html:
```diff
@@ -37,6 +37,7 @@
 
 var fb = document.querySelector(".filebrowser");
 new FileBrowser(fb, "/var/www/html/Music", "Music", "Music");
+new FileBrowser(fb, "/var/www/html/LightShow", "LightShow", "LightShow");
```
This won't work, as it's attaching both FileBrowsers to the same div in the html document.
It doesn't really matter though since this html file is not used.

MagoKimbra committed:
Delete teslausb-www/html/filebrowser.html

MagoKimbra committed:
Update install.sh

MagoKimbra committed:
Update create-backingfiles-partition.sh

MagoKimbra committed:
Update teslausb_setup_variables.conf.sample

MagoKimbra committed:
Update install.sh

MagoKimbra committed:
Update create-backingfiles.sh

MagoKimbra committed:
Update install.sh

MagoKimbra committed:
Update install.sh

MagoKimbra committed:
Update install.sh

MagoKimbra committed:
Update install.sh

MagoKimbra committed:
Update configure-web.sh

MagoKimbra committed:
Update create-backingfiles-partition.sh

MagoKimbra committed:
Update awake_start

tesla_api.py occasionally hangs/blocks long enough for the car to go
to sleep again. Add a timeout so that if the 'ping' doesn't complete
in a timely manner, it is retried.

MagoKimbra committed:
tesla_api: use products URL to list vehicles

Tesla now has an official API (which TeslaUSB likely won't be able
to use) but the old API still partly works with a different URL, so
use that for now.

MagoKimbra committed:
setup: install parted if needed

MagoKimbra committed:
setup: tweak create-backingfiles-partition and add a test

Deduplicate some common code in the partitioning script.
To avoid a future repeat of the recent partitioning script breakage,
add a test to verify its behavior in various scenarios.

MagoKimbra committed:
setup: tweak create-backingfiles-partition and add a test

Deduplicate some common code in the partitioning script.
To avoid a future repeat of the recent partitioning script breakage,
add a test to verify its behavior in various scenarios.

MagoKimbra committed:
Update envsetup.sh

Add optional Lightshow drive

MagoKimbra committed:
www: fix contextmenu for multiple filebrowser instances

MagoKimbra committed:
www: fix contextmenu for multiple filebrowser instances

MagoKimbra committed:
www: drag-to-select improvements for touch  main-dev

MagoKimbra committed:
www: remove unused function

MagoKimbra committed:
www: rework file browser

Combine music and lightshow into one tab, and add boombox too.

marcone commented:
This needed some additional fixes because it wasn't quite working as-is, but it is now merged as 2ad9e08e67ed14946f5d116f1e90d7f2c39a3559 and 1174185b4477af9c24cdd7af52e9b122523e221f
